radix-engine/src/updates/protocol_updates.rs

```diff
@@ -0,0 +1,106 @@
+use super::state_updates::*;
+use crate::{internal_prelude::*, track::StateUpdates};
+use radix_substate_store_interface::interface::SubstateDatabase;
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
+pub enum ProtocolUpdate {
```

Comment:
Seems like a flat enum will get quite large as we add more and more protocol updates. Perhaps this should be something like:

```rust
pub enum ProtocolUpdate {
    AnemoneProtocolUpdate(AnemoneProtocolUpdate),
    BottlenoseProtocolUpdate(BottlenoseProtocolUpdate),
}

pub enum AnemoneProtocolUpdate {
    /// Introduces BLS12-381 and Keccak-256 features.
    Bls12381AndKeccak256,

    /// Exposes second-precision timestamp.
    SecondPrecisionTimestamp,

    /// Increases the math precision with native pool implementations.
    PoolMathPrecisionFix,

    /// Changes the cost associated with validator creation.
    ValidatorCreationFeeFix,

    /// Moves various protocol parameters to state.
    ProtocolParamsToState,
}

pub enum BottlenoseProtocolUpdate {
    /// Exposes a getter method for reading owner role rule.
    OwnerRoleGetter,

    /// Various system patches.
    SystemPatches,

    /// Introduces the account locker blueprint.
    AccountLocker,
}
```

Reply:
I'm not sure if I see the benefits.

I viewed each of the protocol update as an "EIP". The definition of EIP shouldn't depend on the network updates that will include it.

Reply:
Main benefits is just code organization. However, as you mentioned, if some upgrades will be made without a network upgrade then flat enum would make more sense.